pkg/utils/utils.go

```diff
@@ -77,6 +80,13 @@ func Hash(value string, hashLength int) string {
 // This function shortens a name to the maximum length given and uses rest of the string that is too long
 // as hash that gets added to the valid name.
 func ShortenedName(name string, maxLength, hashLength int) string {
+	actualBound := int(math.Min(float64(maxLength), float64(len(name))))
+	// First shorten so if there is a problem it's due to ilegal chars, rahter its legnth
+	if ilegal := validation.IsDNS1123Subdomain(name[:actualBound]); ilegal != nil {
```

Comment:
Suppose you have an illegal name "xxx/yyy", maxLength=5, hashLength=2. In this case actualBound = 5, so you check "xxx/y" and end up in hashing all of it. In the original code, only "xxx" is taken as prefix and the rest is used as hashed suffix of length 2. So, you over-check.